### ISSUE TITLE
 jsdoc/valid-types syntax error in namepath: Foo#bar on @fires/@emits using TypeScript

### ISSUE THREAD

**Issue by yvele** [gajus/eslint-plugin-jsdoc#750]
## Expected behavior

This event name format is supposed to work **using TypeScript**:

```txt
@fires <className>#[event:]<eventName>
```

## Actual behavior

On TypeScript, the tag `@fires` (and `@emits`) does not support writing an event name using the official JSDoc syntax https://jsdoc.app/tags-fires.html and makes the `jsdoc/valid-types` to fail.

Note that the same code works on plain JavaScript code (no TypeScript).

## ESLint Config

```js
module.exports = {
  "env": {
    "browser": true,
    "es2021": true
  },
  "extends": [
    "eslint:recommended",
    "plugin:@typescript-eslint/recommended"
  ],
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 12,
    "sourceType": "module"
  },
  "plugins": [
    "jsdoc",
    "@typescript-eslint"
  ],
  rules : {
    "jsdoc/valid-types" : "error"
  }
};

```

## ESLint sample

```ts
/**
 * Foo.
 *
 * @fires Foo#bar
 */
export default class Foo {

}

```

## Environment

- Node version: v14.17.0
- ESLint version: v7.27.0
- `eslint-plugin-jsdoc` version: v35.1.2

**Comment by simonseyock** [comment]
The name path syntax for `@fires` or `@emit` is not actually a valid typescript syntax. So it is rejected by the parser. In fact these tags are ignored by typescript altogether.

An option would be to instead of `parse(expression, 'typescript')` use `tryParse(expression, ['typescript', 'jsdoc'])`. This should only be done for these specific tags.

**Comment by brettz9** [comment]
Yeah, even though TypeScript currently doesn't have any special handling for it, it silently ignores them in those contexts, so we ought to restore support for it one way or another. I think it may be better if you can support different contexts because AFAIK TypeScript may well begin to support event paths at some point, and then it would no longer be a mode issue but a context question. It is also a context question now in that a tag like `@function` probably shouldn't accept an event namepath as its name (even though it can accept a namepath IIRC).

@yvele : We have updated our type parser to use @simonseyock 's excellent work in adding better and more TypeScript support. but expect a few hiccups like this as we get these kinks ironed out.
